File: iOS/Models/ClientNamePolicy.swift
import Foundation

/// Canonical profile-name presentation/duplicate policy; UUID remains the real identity.
public enum ClientNamePolicy: Sendable {
    public static func cleaned(_ raw: String) -> String {
        raw.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    public static func canonical(_ raw: String) -> String {
        cleaned(raw).folding(options: [.caseInsensitive, .diacriticInsensitive], locale: Locale(identifier: "en_US_POSIX"))
    }

    public static func isValid(_ raw: String) -> Bool {
        !cleaned(raw).isEmpty
    }
}

File: Android/app/src/main/java/com/example/supplementtracker/domain/util/ClientNamePolicy.kt
package com.example.supplementtracker.domain.util

import java.text.Normalizer
import java.util.Locale

/** Canonical profile-name presentation/duplicate policy; UUID remains the real identity. */
object ClientNamePolicy {
    fun cleaned(raw: String): String = raw.trim()

    fun canonical(raw: String): String {
        val decomposed = Normalizer.normalize(cleaned(raw), Normalizer.Form.NFD)
        return decomposed.replace(Regex("\\p{M}+"), "").lowercase(Locale.ROOT)
    }

    fun isValid(raw: String): Boolean = cleaned(raw).isNotEmpty()
}
